src/java/org/apache/cassandra/io/sstable/format/SSTableReader.java

```diff
@@ -1199,7 +1219,15 @@ else if (samplingLevel < indexSummary.getSamplingLevel())
             }
 
             // Always save the resampled index
-            saveSummary(newSummary);
+            tidy.global.writeLock.lock();
+            try
+            {
+                saveSummary(newSummary);
+            }
+            finally
+            {
+                tidy.global.writeLock.unlock();
+            }
```

Comment:
What do you think about having a `runWithWriteLock(Runnable)` or something similar on `GlobalTidy`? I'm guessing we're not concerned about the overhead of crating a lambda, etc.

Reply:
+1